src/main/java/org/kaazing/netx/ws/internal/WebSocketInputStateMachine.java

```diff
@@ -56,143 +51,73 @@
             stateMachine[state.ordinal()][ERROR.ordinal()] = CLOSED;
         }
 
-        stateMachine[START.ordinal()][RECEIVED_UPGRADE_RESPONSE.ordinal()] = OPEN;
+        stateMachine[START.ordinal()][RECEIVE_UPGRADE_RESPONSE.ordinal()] = OPEN;
 
-        stateMachine[OPEN.ordinal()][RECEIVED_PING_FRAME.ordinal()] = OPEN;
-        stateMachine[OPEN.ordinal()][RECEIVED_PONG_FRAME.ordinal()] = OPEN;
-        stateMachine[OPEN.ordinal()][RECEIVED_CLOSE_FRAME.ordinal()] = OPEN;
-        stateMachine[OPEN.ordinal()][RECEIVED_BINARY_FRAME.ordinal()] = OPEN;
-        stateMachine[OPEN.ordinal()][RECEIVED_TEXT_FRAME.ordinal()] = OPEN;
+        stateMachine[OPEN.ordinal()][RECEIVE_PING_FRAME.ordinal()] = OPEN;
+        stateMachine[OPEN.ordinal()][RECEIVE_PONG_FRAME.ordinal()] = OPEN;
+        stateMachine[OPEN.ordinal()][RECEIVE_CONTINUATION_FRAME.ordinal()] = OPEN;
+        stateMachine[OPEN.ordinal()][RECEIVE_CLOSE_FRAME.ordinal()] = OPEN;
+        stateMachine[OPEN.ordinal()][RECEIVE_BINARY_FRAME.ordinal()] = OPEN;
+        stateMachine[OPEN.ordinal()][RECEIVE_TEXT_FRAME.ordinal()] = OPEN;
 
 
         STATE_MACHINE = stateMachine;
     }
 
-    private final FrameFactory frameFactory;
+    private static final WebSocketInputStateMachine INSTANCE = new WebSocketInputStateMachine();
 
-    public WebSocketInputStateMachine() {
-        frameFactory = FrameFactory.newInstance(8192);
+    private WebSocketInputStateMachine() {
     }
 
-    public void start(WsURLConnectionImpl connection) {
-        connection.setInputState(WebSocketState.START);
-    }
-
-    public void processBinary(final WsURLConnectionImpl connection,
-                              final Data dataFrame,
-                              final WebSocketFrameConsumer terminalConsumer)
-            throws IOException {
-        WebSocketExtensionSpi sentinel = new WebSocketExtensionSpi() {
-            {
-                onBinaryFrameReceived = terminalConsumer;
-            }
-        };
-        WebSocketContext context = connection.getContext(sentinel, false);
-        WebSocketState state = connection.getInputState();
-
-        switch (state) {
-        case OPEN:
-            transition(connection, WebSocketTransition.RECEIVED_BINARY_FRAME);
-            context.onBinaryReceived(dataFrame);
-            break;
-        default:
-            transition(connection, WebSocketTransition.ERROR);
-            context.onError(format("Invalid state %s to be receiving a BINARY frame", state));
-        }
+    public static WebSocketInputStateMachine instance() {
+        return INSTANCE;
```

Comment:
Do we need this static accessor?  Perhaps it is already scoped by `WsURLConnectionHelper` instance?